engines/scc_proxy/lib/scc_proxy/engine.rb

```diff
@@ -68,7 +68,13 @@ def get_instance_id(params)
         nil,
         params['instance_data']
       )
-      instance_id_key = INSTANCE_ID_KEYS[params['hwinfo']['cloud_provider'].downcase.to_sym]
+      id_key = params['hwinfo']['cloud_provider'].downcase.to_sym
+      unless INSTANCE_ID_KEYS.key?(id_key)
+        error_message = 'Unknown cloud provider'
+        Rails.logger.error error_message
+        raise ActionController::TranslatedError.new(error_message)
```

Comment:
Why? I am unaware of any conversation at the business level where a decision was made that 3rd party CSPs that run their own update infrastructure are not allowed to offer BYOS pass through.

Reply:
if the CSP is unknown we would not have a valid value for `system_token`

do we not care about that ?

Reply:
I do not understand what the `system_token` has to do with this change. Nor what `system_token` has to do with now breaking functionality for 3rd party CSPs that previously probably worked when `disable_system_tokens: true` is set in the rmt.conf file.